specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutions.json

```diff
@@ -481,86 +483,106 @@
           "type": "string",
           "description": "The recommendation type.",
           "enum": [
-            "OpenPortsOnDevice",
-            "PermissiveFirewallPolicy",
-            "PermissiveFirewallRuleInput",
-            "PermissiveFirewallRuleOut",
-            "OperationSystemNotValid",
-            "UnutilizedMessagesFromAgent",
-            "SecurityTwinConfigurationNotOptimal",
-            "SecurityTwinConfigurationConflict",
-            "IdenticalAuthenticationCredentials",
-            "DenyDefaultIpPolicy",
-            "TooLargeIPRange",
-            "EnableDiagnosticsLog"
+            "IoT_ACRAuthentication",
+            "IoT_AgentSendsUnutilizedMessages",
+            "IoT_Baseline",
+            "IoT_EdgeHubMemOptimize",
+            "IoT_EdgeLoggingOptions",
+            "IoT_InconsistentModuleSettings",
+            "IoT_InstallAgent",
+            "IoT_IPFilter_DenyAll",
```

Comment:
I'm not completely sure that "_" is a permissible value in each of the languages the swagger is used to generate. Might be easier to go with "IPFilterDenyAll" and "IPFilterPermissiveRule"

Reply:
This was reviewed with our PMs & technical writers.
This field contains the type of the recommendation which is a code that the customer can use in order to view specific recommendation. there is also fields with the description that explains the customer what each recommendation means.
it should be the same value as we generates internally in our system and changing it is a lot of work.
All swagger tests were passed.

Reply:
Makes sense. So I take it you've never published an SDK with these old values? @nschonni is reviewing from the SDK side, so if he's good with it then it's fine by me.

Reply:
@antmarti-microsoft sorry, I'm actually not an MS employee :wink:

Reply:
Ah sorry, my bad! In that case @shahabhijeet it looks like you've been assigned to this PR - please could you take a look?